app/lib/actions/katello/repository/update.rb

```diff
@@ -42,6 +43,24 @@ def plan(root, repo_params)
                                        content_type: repository.content_type,
                                        label: content.label,
                                        gpg_url: repository.yum_gpg_key_url)
+
+            if root.previous_changes.key?('os_versions')
+              # If Restrict to OS Version is changing _from_ or _to_ 'rhel-6'...
+              if root.previous_changes['os_versions'].any? { |ch| ch.include?('rhel-6') }
+                # ...force regeneration of entitlement certs on RHEL 6.8 and older
+                Rails.logger.info("Looking for RHEL 6.8 and older clients")
+                cp_consumers = ::Katello::Host::ContentFacet
+                  .joins(host: :operatingsystem)
+                  .where("operatingsystems.major = '6' AND operatingsystems.minor < '9' AND operatingsystems.name = 'RedHat'")
+                  .map(&:uuid)
+                # TODO: make this an action
+                Rails.logger.info("Marking entitlement certs dirty for #{pluralize(cp_consumers.length, 'client')}")
+                cp_consumers.each do |uuid|
+                  ::Katello::Resources::Candlepin::Entitlement.regenerate_entitlement_certificates_for_consumer(uuid, true)
+                end
```

Comment:
If we decide to go with this approach I'll make this a Dynflow action.

Reply:
+1 and it should spawn off another task, not run within this one.  We do this for example here:  https://github.com/Katello/katello/blob/master/app/lib/actions/katello/repository/update.rb#L62